File: sample/src/main/java/de/blox/graphview/sample/Algorithms/FruchtermanReingoldActivity.java
package de.blox.graphview.sample.Algorithms;

import de.blox.graphview.Graph;
import de.blox.graphview.GraphView;
import de.blox.graphview.Node;
import de.blox.graphview.energy.FruchtermanReingoldAlgorithm;
import de.blox.graphview.sample.GraphActivity;

public class FruchtermanReingoldActivity extends GraphActivity {

    @Override
    public Graph createGraph() {
        final Graph graph = new Graph();
        final Node a = new Node(getNodeText());
        final Node b = new Node(getNodeText());
        final Node c = new Node(getNodeText());
        final Node d = new Node(getNodeText());
        final Node e = new Node(getNodeText());
        final Node f = new Node(getNodeText());
        final Node g = new Node(getNodeText());
        final Node h = new Node(getNodeText());

        graph.addEdge(a, b);
        graph.addEdge(a, c);
        graph.addEdge(a, d);
        graph.addEdge(c, e);
        graph.addEdge(d, f);
        graph.addEdge(f, c);
        graph.addEdge(g, c);
        graph.addEdge(h, g);

        return graph;
    }

    @Override
    public void setLayout(GraphView view) {
        view.setLayout(new FruchtermanReingoldAlgorithm(1000));
    }
}

File: sample/src/main/java/dev/bandb/graphview/sample/algorithms/FruchtermanReingoldActivity.kt
package dev.bandb.graphview.sample.algorithms

import dev.bandb.graphview.decoration.edge.ArrowEdgeDecoration
import dev.bandb.graphview.graph.Graph
import dev.bandb.graphview.graph.Node
import dev.bandb.graphview.layouts.energy.FruchtermanReingoldLayoutManager
import dev.bandb.graphview.sample.GraphActivity

class FruchtermanReingoldActivity : GraphActivity() {

    public override fun setLayoutManager() {
        recyclerView.layoutManager = FruchtermanReingoldLayoutManager(this, 1000)
    }

    public override fun setEdgeDecoration() {
        recyclerView.addItemDecoration(ArrowEdgeDecoration())
    }

    public override fun createGraph(): Graph {
        val graph = Graph()
        val a = Node(nodeText)
        val b = Node(nodeText)
        val c = Node(nodeText)
        val d = Node(nodeText)
        val e = Node(nodeText)
        val f = Node(nodeText)
        val g = Node(nodeText)
        val h = Node(nodeText)
        graph.addEdge(a, b)
        graph.addEdge(a, c)
        graph.addEdge(a, d)
        graph.addEdge(c, e)
        graph.addEdge(d, f)
        graph.addEdge(f, c)
        graph.addEdge(g, c)
        graph.addEdge(h, g)
        return graph
    }
}